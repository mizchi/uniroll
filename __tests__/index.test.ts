import "isomorphic-unfetch";

import { compileToString } from "../index";
import assert from "assert";

it("compile (no deps)", async () => {
  const out = await compileToString({
    entry: "/index.js",
    files: {
      "/index.js": `export default 1;`
    }
  });
  expect(out).toMatchSnapshot();
});

it("compile with relational path", async () => {
  const out = await compileToString({
    entry: "/index.js",
    files: {
      "/index.js": `export { default as foo } from "./foo"`,
      "/foo.js": "export default 2;"
    }
  });
  expect(out).toMatchSnapshot();
});

it("compile ts", async () => {
  const out = await compileToString({
    entry: "/index.ts",
    files: {
      "/index.ts": `const foo: number = 3; export default foo as number;`,
      "/tsconfig.json": `{ "compilerOptions": { "target": "es5" } }`
    }
  });
  expect(out).toMatchSnapshot();
});

it("compile tsx", async () => {
  const out = await compileToString({
    entry: "/index.tsx",
    files: {
      "/index.tsx": `const h = () => {};export default <div>1</div>`,
      "/tsconfig.json": `{ "compilerOptions": { "target": "es5", "jsx": "react", "jsxFactory": "h" } }`
    }
  });
  expect(out).toMatchSnapshot();
});

it("compile with rollupConfig", async () => {
  const out = await compileToString({
    entry: "/index.js",
    files: {
      "/index.js": `export default 1;`
    },
    rollupConfig: {
      name: "_1",
      format: "umd"
    }
  });
  assert(out.includes("define.amd"));
  expect(out).toMatchSnapshot();
});

it("compile with minify", async () => {
  const out = await compileToString({
    entry: "/index.js",
    files: {
      "/index.js": `export default 1;`
    }
  });

  const minified = await compileToString({
    entry: "/index.js",
    minify: true,
    files: {
      "/index.js": `export default 1;`
    }
  });

  assert(out.length > minified.length);
  expect(minified).toMatchSnapshot();
});
