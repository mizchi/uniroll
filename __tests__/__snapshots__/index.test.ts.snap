// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compile (no deps) 1`] = `
"var index = 1;

export default index;
"
`;

exports[`compile ts 1`] = `
"Object.defineProperty(exports, \\"__esModule\\", { value: true });
var foo = 3;
exports.default = foo;
"
`;

exports[`compile tsx 1`] = `
"Object.defineProperty(exports, \\"__esModule\\", { value: true });
var h = function () { };
exports.default = h();
"
`;

exports[`compile with minify 1`] = `"var index=1;export default index;"`;

exports[`compile with relational path 1`] = `
"var foo = 2;

export { foo };
"
`;

exports[`compile with rollupConfig 1`] = `
"(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
	typeof define === 'function' && define.amd ? define(factory) :
	(global = global || self, global._1 = factory());
}(this, (function () { 'use strict';

	var index = 1;

	return index;

})));
"
`;
